Runtime/Interaction/StaticMeshCollidersManager.cs

```diff
@@ -0,0 +1,223 @@
+        public float OffsetX => offsetX;
+        public float OffsetY => offsetY;
+        public int MeshXResolution => meshXResolution;
+        public int MeshYResolution => meshYResolution;
+        public Dictionary<Vector2Int, Collider> RawCoordsToCollider => rawCoordsToCollider;
+
+        private void Update()
+        {
+            if (!generatedColliders)
+            {
+                return;
+            }
+            
+            UpdateColliderPositions();
+            
+            //FIXME: Debug Code
+            // rectifiedVertexDS.transform.localPosition = vertices[remappedVertices[id]];
+        }
+
+        private void OnDestroy()
+        {
+            vertices_native.Dispose();
+            normals_native.Dispose();
+            remapped_vertices_data.Dispose();
+            colliderObjects.Dispose();
+        }
+
+        public List<Collider> SetupColliders(SkinnedMeshRenderer keyboardMesh, HPUIStaticContinuousInteractable hpuiStaticContinuousInteractable)
+        {
+            targetMesh = keyboardMesh;
+            tempMesh = new Mesh(); 
+            targetMesh.BakeMesh(tempMesh, true);
+            
+            if (vertexRemapData == null)
+            {
+                throw new ArgumentException("Missing Vertex Remap Data Asset! Create a new one or provide an existing one!");
+            }
+            
+            tempMesh.GetVertices(vertices);
+            tempMesh.GetNormals(normals);
+            
+            remapped_vertices_data = new NativeArray<int>(vertexRemapData, Allocator.Persistent);
+            
+            meshXResolution = hpuiStaticContinuousInteractable.MeshXResolution;
+            if (vertices.Count % meshXResolution != 0)
+            {
+                throw new Exception("Total vertex count doesn't divide properly with X mesh resolution!");
+            }
+            meshYResolution = vertices.Count / meshXResolution;
+            
+            xWidth = Vector3.Distance(vertices[remapped_vertices_data[0]], vertices[remapped_vertices_data[1]]);
+            yWidth = Vector3.Distance(vertices[remapped_vertices_data[0]], vertices[remapped_vertices_data[meshXResolution]]);
+            
+            offsetX = xWidth * meshXResolution * 0.5f;
+            offsetY = yWidth * meshYResolution * 0.5f;
+            Transform[] colliderTransforms = new Transform[vertices.Count];
+            Transform meshTransform = targetMesh.gameObject.transform;
+            InitCollidersState(meshTransform, colliderTransforms);
+            colliderObjects = new TransformAccessArray(colliderTransforms);
+            vertices_native = new NativeArray<Vector3>(vertices.ToArray(), Allocator.Persistent);
+            normals_native = new NativeArray<Vector3>(normals.ToArray(), Allocator.Persistent);
+            generatedColliders = true;
+
+            return colliderCoords.Keys.ToList();
+        }
+
+        private void InitCollidersState(Transform meshTransform, Transform[] colliderTransforms)
```

Comment:
I would generally discourage this. Unless there is a reason to abstract it, I would inline this.

Reply:
done